integration-tests/src/integrationTest/org/apache/cassandra/sidecar/routes/CompactionStatsIntegrationTest.java

```diff
@@ -0,0 +1,372 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.routes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpResponseExpectation;
+import io.vertx.ext.web.client.HttpResponse;
+import org.apache.cassandra.sidecar.common.response.CompactionStatsResponse;
+import org.apache.cassandra.sidecar.common.response.data.ActiveCompactionEntry;
+import org.apache.cassandra.sidecar.testing.QualifiedName;
+import org.apache.cassandra.sidecar.testing.SharedClusterSidecarIntegrationTestBase;
+
+import static org.apache.cassandra.testing.TestUtils.DC1_RF1;
+import static org.apache.cassandra.testing.TestUtils.TEST_KEYSPACE;
+import static org.apache.cassandra.testing.TestUtils.TEST_TABLE_PREFIX;
+import static org.apache.cassandra.testing.utils.AssertionUtils.getBlocking;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Integration tests for the Compaction Statistics API endpoint.
+ * 
+ * <p>Tests the compaction statistics functionality by generating large-scale SSTable data across
+ * multiple tables, triggering concurrent compactions, and validating API responses in both active
+ * and completed compaction scenarios.</p>
+ * 
+ * <h2>Testing Mechanism</h2>
+ * <ul>
+ *   <li><strong>Data Setup:</strong> Creates 5 tables with 100 SSTables each (500 total, 1000 records per SSTable)</li>
+ *   <li><strong>Concurrency:</strong> Triggers compactions simultaneously using separate threads</li>
+ *   <li><strong>Active Detection:</strong> Polls API immediately with retry logic to catch in-progress compactions</li>
+ *   <li><strong>Comprehensive Validation:</strong> Validates all response fields including statistics, rates, 
+ *       pending tasks, and detailed active compaction data when available</li>
+ *   <li><strong>Dual Success:</strong> Test passes whether active compactions are captured or completed</li>
+ * </ul>
+ */
+class CompactionStatsIntegrationTest extends SharedClusterSidecarIntegrationTestBase
```

Comment:
we should merge this test class with `org.apache.cassandra.sidecar.routes.CassandraStatsIntegrationTest`